mono/metadata/reflection.c

```diff
@@ -10715,19 +10715,23 @@ mono_reflection_register_with_runtime (MonoReflectionType *type)
  * LOCKING: Assumes the loader lock is held.
  */
 static MonoMethodSignature*
-parameters_to_signature (MonoImage *image, MonoArray *parameters) {
-	MonoError error;
+parameters_to_signature (MonoImage *image, MonoArray *parameters, MonoError *error) {
 	MonoMethodSignature *sig;
 	int count, i;
 
+	mono_error_init (error);
+
 	count = parameters? mono_array_length (parameters): 0;
 
 	sig = (MonoMethodSignature *)image_g_malloc0 (image, MONO_SIZEOF_METHOD_SIGNATURE + sizeof (MonoType*) * count);
 	sig->param_count = count;
 	sig->sentinelpos = -1; /* FIXME */
 	for (i = 0; i < count; ++i) {
-		sig->params [i] = mono_type_array_get_and_resolve (parameters, i, &error);
-		mono_error_raise_exception (&error); /* FIXME don't raise here */
+		sig->params [i] = mono_type_array_get_and_resolve (parameters, i, error);
+		if (!is_ok (error)) {
+			image_g_free (image, sig);
```

Comment:
don't we risk to free that image in cases we shouldn't? it seems weird to me to do it here.